packages/app/src/pages/PoolPage/PoolPage.test.tsx

```diff
@@ -1,5 +1,61 @@
+import "cross-fetch/polyfill";
+import { fireEvent, screen, waitFor } from "@fuels-ui/test-utils";
+
+import { createWallet } from "../WelcomePage/WalletPage.test";
+
+export const openPoolList = async () => {
+  await waitFor(async () => {
+    const poolMenuBtn = await screen.findByTestId("poolMenuBtn");
+    expect(poolMenuBtn).toBeInTheDocument();
+    fireEvent.click(poolMenuBtn);
+  });
+
+  await waitFor(async () => {
+    const headerAddLiquidityBtn = await screen.findByTestId(
+      "headerAddLiquidityBtn"
+    );
+    expect(headerAddLiquidityBtn).toBeInTheDocument();
+  });
+};
+
+export const openAddLiquidity = async () => {
+  await waitFor(async () => {
+    const headerAddLiquidityBtn = await screen.findByTestId(
+      "headerAddLiquidityBtn"
+    );
+    expect(headerAddLiquidityBtn).toBeInTheDocument();
+    fireEvent.click(headerAddLiquidityBtn);
+  });
+
+  await waitFor(async () => {
+    const addLiquiditySubmitBtn = await screen.findByTestId(
+      "addLiquiditySubmitBtn"
+    );
+    expect(addLiquiditySubmitBtn).toBeInTheDocument();
+  });
+};
+
 describe("PoolPage", () => {
-  it("pass tests", () => {
-    expect(true);
+  it("should validate pool flow", async () => {
+    await createWallet();
```

Comment:
I really don't think you need to create a wallet and make all steps Welcome-related necessary steps here, you're testing Pools page, everything that isn't related to Pool, you can mock. If you always make all steps on each scenario, our suite will take so much to run.

Like I did in the old tests you removed:

```js
const { privateKey } = Wallet.generate({ provider: FUEL_PROVIDER_URL });
const wallet = new Wallet(privateKey, FUEL_PROVIDER_URL);

jest.mock("../../hooks/useWallet", () => ({
  useWallet: jest.fn(() => wallet),
}));

jest.mock("../../hooks/useContract.ts", () => ({
  useContract: () => ExchangeContractAbi__factory.connect(CONTRACT_ID, wallet),
}));
```

Reply:
now we have `App.test.tsx` which has a general end-to-end test of the application (with nothing mocked). This is done to test the whole app, instead of small tests of components. For now it's done until the pool part only, but it's nice to continue the work and test the whole app.

we also have the specific tests like `PoolPage.test.tsx`, `WelcomePage.test.tsx`. which mock data in the way you proposed and quickly test small things in the pages